src/FS.Embedded.Manifest.Task/GenerateEmbeddedResourcesManifest.cs

```diff
@@ -0,0 +1,91 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Xml;
+using System.Xml.Linq;
+using Microsoft.Build.Framework;
+
+namespace Microsoft.Extensions.FileProviders.Embedded.Manifest.Task
+{
+    /// <summary>
+    /// Task for generating a manifest file out of the embedded resources in an
+    /// assembly.
+    /// </summary>
+    public class GenerateEmbeddedResourcesManifest : Microsoft.Build.Utilities.Task
+    {
+        private const string LogicalName = "LogicalName";
+        private const string ManifestResourceName = "ManifestResourceName";
+        private const string TargetPath = "TargetPath";
+
+        [Required]
+        public ITaskItem[] EmbeddedFiles { get; set; }
+
+        [Required]
+        public string ManifestFile { get; set; }
+
+        /// <inheritdoc />
+        public override bool Execute()
+        {
+            var processedItems = CreateEmbeddedItems(EmbeddedFiles);
+
+            var manifest = BuildManifest(processedItems);
+
+            var document = manifest.ToXmlDocument();
+
+            var settings = new XmlWriterSettings()
+            {
+                Encoding = Encoding.UTF8
+            };
+
+            using (var xmlWriter = GetXmlWriter(settings))
+            {
+                document.WriteTo(xmlWriter);
+            }
+
+            return true;
+        }
+
+        protected virtual XmlWriter GetXmlWriter(XmlWriterSettings settings)
+        {
+            return XmlWriter.Create(ManifestFile, settings);
+        }
+
+        public EmbeddedItem[] CreateEmbeddedItems(params ITaskItem[] items) =>
+            items.Select(er => new EmbeddedItem
+            {
+                ManifestFilePath = GetManifestPath(er),
+                AssemblyResourceName = GetAssemblyResourceName(er)
+            }).ToArray();
+
+        private string GetManifestPath(ITaskItem er) => string.Equals(er.GetMetadata(LogicalName), er.GetMetadata(ManifestResourceName)) ?
+            er.GetMetadata(TargetPath) :
+            NormalizePath(er.GetMetadata(LogicalName));
+
+        private string GetAssemblyResourceName(ITaskItem er) => string.Equals(er.GetMetadata(LogicalName), er.GetMetadata(ManifestResourceName)) ?
+            er.GetMetadata(ManifestResourceName) :
+            er.GetMetadata(LogicalName);
+
+        private string NormalizePath(string path) => Path.DirectorySeparatorChar == '\\' ?
+            path.Replace("/", "\\") : path.Replace("\\", "/");
+
+        public Manifest BuildManifest(EmbeddedItem[] processedItems)
```

Comment:
Minor: would be nice to have all the public methods before private ones (or private methods in the bottom)